### ISSUE TITLE
error on two enums with the same variant name

### ISSUE THREAD

**Issue by dsilvasc** [shepmaster/snafu#206]
Given two error types with a common variant name:

```rust
#[derive(Debug, Snafu)]
pub enum CreateRoomError {
  #[snafu(display("Room name already exists"))]
  RoomNameExists,
  #[snafu(display("Error calling DynamoDB: {}", error))]
  DynamoError { error: RusotoError<PutItemError> },
}

#[derive(Debug, Snafu)]
pub enum GetRoomError {
  #[snafu(display("Error calling DynamoDB: {}", error))]
  DynamoError { error: RusotoError<GetItemError> },
}
```

This produces a compile error:

```
error[E0428]: the name `DynamoError` is defined multiple times
  --> graphql/src/dynamo.rs:29:17
   |
21 | #[derive(Debug, Snafu)]
   |                 ----- previous definition of the type `DynamoError` here
...
29 | #[derive(Debug, Snafu)]
   |                 ^^^^^ `DynamoError` redefined here
   |
   = note: `DynamoError` must be defined only once in the type namespace of this module
```

Looks like Snafu generates this:

```rust
    #[doc = "SNAFU context selector for the `CreateRoomError::DynamoError` variant"]
    struct DynamoError<__T0> {
        #[allow(missing_docs)]
        error: __T0,
    }
```

Would it be possible to namespace or prefix that generated struct name?

**Comment by shepmaster** [comment]
This sounds like a duplicate of #188. For right now, you can do some workarounds by placing the error in a module yourself and adjusting the visibility. 

Does this seem reasonable to close as a duplicate?

**Comment by dsilvasc** [comment]
You're right -- sorry I missed that one.  Closing this issue in favor of it.
